lib/rucio/core/did.py

```diff
@@ -1833,3 +1841,171 @@ def list_archive_content(scope, name, session=None):
+    :param payload: Any message to be stored along with the event.
+    :param session: The database session in use.
+    """
+    try:
+        dids = session.query(models.DidsFollowed).filter_by(scope=scope, name=name).all()
+
+        for did in dids:
+            # Create a new event using teh specified parameters.
+            new_event = models.FollowEvents(scope=scope, name=name, account=did.account,
+                                            did_type=did.did_type, event_type=event_type, payload=payload)
+            new_event.save(session=session, flush=False)
+
+        session.flush()
+    except IntegrityError as error:
+        raise exception.RucioException(error.args)
+
+
+@read_session
+def create_reports(total_workers, worker_number, session=None):
+    """
+    Create a summary report of the events affecting a dataset, for its followers.
+
+    :param session: The database session in use.
+    """
+    # Query the FollowEvents table
+    query = session.query(models.FollowEvents)
+
+    # Use hearbeat mechanism to select a chunck of events based on the hased account
+    if total_workers > 0:
+        if session.bind.dialect.name == 'oracle':
+            bindparams = [bindparam('worker_number', worker_number),
+                          bindparam('total_workers', total_workers)]
+            query = query.filter(text('ORA_HASH(account, :total_workers) = :worker_number', bindparams=bindparams))
+        elif session.bind.dialect.name == 'mysql':
+            query = query.filter(text('mod(md5(account), %s) = %s' % (total_workers + 1, worker_number)))
+        elif session.bind.dialect.name == 'postgresql':
+            query = query.filter(text('mod(abs((\'x\'||md5(account))::bit(32)::int), %s) = %s' % (total_workers + 1, worker_number)))
+
+    try:
+        events = query.order_by(models.FollowEvents.created_at).all()
+        # If events exist for an account then create a report.
+        if events:
+            body = '''
+                Hello,
+                This is an auto-generated report of the events that have affected the datasets you follow.
+
+                '''
+
+            for i, event in enumerate(events):
+                # Add each event to the message body.
+                body += "{}. Dataset: {} Event: {}\n".format(i + 1, event.name, event.event_type)
+                if event.payload:
+                    body += "Message: {}\n".format(event.payload)
+                body += "\n"
+            body += "Thank You."
+            # Get the email associated with the account.
+            email = session.query(models.Account.email).filter_by(account=event.account)
+            add_message('email', {'to': email,
+                                  'subject': 'Report of affected dataset(s)',
+                                  'body': body})
+
+            # Clean up the events after creating the report
+            session.query(models.FollowEvents).filter_by(account=event.account).\
```

Comment:
I don't think this deletion works, since you only processed a partition of the events of this account, but you are deleting all of them. @cserf suggested to delete the processed events when sending them.